src/TraceEvent/Ctf/CtfTracing.Tests/CtfTraceTests.cs

```diff
@@ -133,5 +138,33 @@ public void LTTng_GCStartStopEvents()
                 }
             }
         }
+
+        [Fact]
+        public void LTTng_KernelEvents()
+        {
+            // test trace files with kernel events only, with clr events only or with both types of events presents
+            string[] files = new string[] { "kernel-only.trace.zip", "clr-only.trace.zip", "kernel-clr.trace.zip" };
+            foreach(string file in files)
+            {
+                string path = Path.Combine(TestDataDirectory, file);
+                using (CtfTraceEventSource ctfSource = new CtfTraceEventSource(path))
+                {
+                    var kernelParser = new LinuxKernelEventParser(ctfSource);
+
+                    var procStartList = new List<TraceEvent>();
+                    var gcList = new List<TraceEvent>();
+                    kernelParser.ProcessStart += delegate (ProcessStartTraceData data)
+                    {
+                        procStartList.Add(data.Clone());
```

Comment:
This looks great.  Can I recommend that instead of cloning the items and adding them to a list that you count the number of each event and then assert that it matches the expected count at the end?  This adds just a bit more robustness.

Reply:
I was thinking about what asserts it can do and this sounds like a good one!